eras/shelley/impl/src/Cardano/Ledger/Shelley/Rules/Utxow.hs

```diff
@@ -406,23 +405,16 @@ validateFailedNativeScripts (ScriptsProvided scriptsProvided) tx = do
 {-  sNeeded := scriptsNeeded utxo tx                             -}
 {-  sProvided := Map.keysSet (tx ^. witsTxL . scriptTxWitsL)     -}
 validateMissingScripts ::
-  EraPParams era =>
-  PParams era ->
   ShelleyScriptsNeeded era ->
   ScriptsProvided era ->
   Test (ShelleyUtxowPredFailure era)
-validateMissingScripts pp (ShelleyScriptsNeeded sNeeded) scriptsprovided =
-  if HardForks.missingScriptsSymmetricDifference (pp ^. ppProtocolVersionL)
```

Comment:
I wonder if we are sure that this case hasn't happened on mainnet -  in theory we can remove these checks only if that is indeed the case. 
But since there's a ticket about it which doesn't mention anything about testing, I suppose it's safe to  assume that it's safe to do. 
Maybe we can ask @disassembler  at the meeting today.

Reply:
Summarizing our discussion from the meeting today:

1. The protocol version in `PParams` will change as we process blocks from genesis through to the present.
2. The effect of this PR is to change the specific predicate failures that result from an invalid transaction, but not to change the validity of a given transaction.
3. We _think_ that there will be no invalid transactions in the history before Babbage, but @disassembler thought that it might be possible with some smart contracts. This could be tested, if necessary.
4. We _think_ that the specific predicate failures don't matter, and only the validity matters.
5. We'll wait for @lehins to give an opinion on this next week.

@teodanciu Please add a comment if this isn't accurate.